### PR TITLE
Remove direct reference from System.Runtime.WindowsRuntime to System.Private.CoreLib

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Update System.Runtime.WindowsRuntime to not directly reference System.Private.CoreLib.

- Move `Internal.Runtime.InteropServices.WindowsRuntime.ExceptionSupport` in S.P.CL to `System.Runtime.InteropServices.WindowsRuntime.ExceptionSupport` in S.R.WR
- Expose `System.Runtime.InteropServices.WindowsRuntime.WindowsRuntimeMarshal.GetUniqueObjectForIUnknownWithoutUnboxing(System.IntPtr)` and `System.Runtime.InteropServices.WindowsRuntime.EventRegistrationTokenTable<T>.RemoveEventHandler(System.Runtime.InteropServices.WindowsRuntime.EventRegistrationToken, T)` from System.Runtime.InteropServices.WindowsRuntime facade assembly.
- Expose `System.Runtime.InteropServices.WindowsRuntime.WindowsRuntimeMarshal.ReportUnhandledError` from System.Runtime.InteropServices.WindowsRuntime facade assembly.
- Redesign `System.Resources.WindowsRuntimeResourceManager` to be called from System.Private.CoreLib exclusively through reflection (see coreclr PR for more information)
- Add local implementation of `Internal.Threading.Tasks.AsyncCausalitySupport` that calls the implementation in System.Private.CoreLib via reflection.
- Expose `System.Runtime.InteropServices.WindowsRuntime.WindowsRuntimeMarshal.ReportUnhandledError`.

cc: @terrajobst for API review of the newly exposed APIs.

Depends on dotnet/coreclr#26443

### CODE REVIEW

File: src/System.Runtime.WindowsRuntime/src/System/InternalHelpers.cs [hunk 1]
```diff
@@ -14,3 +17,81 @@ internal static void SetErrorCode(this Exception ex, int code)
...
+        private static Func<bool> _loggingOn;
+        private static Action<Task, string> _traceOperationCreation;
+        private static Action<Task> _traceOperationCompletedSuccess;
+        private static Action<Task> _traceOperationCompletedError;
+
+// Initialize all static fields in 'AsyncCausalitySupport' when those fields are declared and remove the explicit static constructor
+#pragma warning disable CA1810
+        static AsyncCausalitySupport()
+        {
+            Type privateType = typeof(object).Assembly.GetType("Internal.Threading.Tasks.AsyncCausalitySupport");
```
Comment (jkotas) [inline src/System.Runtime.WindowsRuntime/src/System/InternalHelpers.cs]: Calling this via reflection is not an improvement.

Comment (jkoritzinsky) [reply src/System.Runtime.WindowsRuntime/src/System/InternalHelpers.cs]: I'd also be fine with surfacing this through the `System.Runtime.InteropServices.WindowsRuntime` facade if we felt that was a better choice.

**Comment by terrajobst** [comment]
> cc: @terrajobst for API review of the newly exposed APIs.

Let's mangle API reviews with PRs. Please file a separate issue and describe why these new APIs are needed. We don't review PRs during API reviews.

**Comment by jkoritzinsky** [comment]
Once we figure out what we’re doing with this PR and the CoreCLR PR, I’ll split out the API review changes into another issue.

**Comment by maryamariyan** [comment]
@jkoritzinsky thanks for the PR. We're trying to reduce the backlog of old PRs so we can focus on being responsive on the active ones. I am going to close this one for now especially since it has not been updated in the past month.

Please do reopen when you're ready to pick this up again, the API review is ready and when the timeline for it getting merged has become clear.

The two repos (coreclr and corefx) will be getting merged soon and that will cause all open PRs (active or inactive) to get closed anyway, and you may loose track of these when that happens. So my recommendation is to close coreclr PR as well and reopen when they both get ready.

**Pull request closed without merging**